apps/settings/test/unit/modules/bluetooth/bluetooth_adapter_manager_test.js

```diff
@@ -0,0 +1,90 @@
+'use strict';
+
+suite('BluetoothAdapterManager', function() {
+  var adapterManager;
+
+  setup(function(done) {
+    var modules = [
+      'modules/bluetooth/bluetooth_adapter_manager'
+    ];
+
+    var map = {
+      'modules/bluetooth': {
+        'modules/navigator/mozBluetooth': 'MockNavigatorBluetooth'
+      }
+    };
+
+    this.MockNavigatorBluetooth = {
+      defaultAdapter: {},
+      callbacks: [],
+      addEventListener: function(eventName, callback) {
+        this.callbacks[eventName] = callback;
+      }
+    };
+
+    define('MockNavigatorBluetooth', function() {
+      return this.MockNavigatorBluetooth;
+    }.bind(this));
+
+    testRequire(modules, map, function(AdapterManager) {
+      adapterManager = AdapterManager;
+      done();
+    });
+  });
+
+  suite('_init > ', function() {
+    setup(function() {
+      this.sinon.stub(adapterManager, '_watchMozBluetoothOnattributechanged');
+      this.sinon.stub(adapterManager, '_initDefaultAdapter');
+    });
+
+    test('mozBluetooth onattributechanged should be watched, ' +
+         'defaultAdapter should be inited ', function() {
+      adapterManager._init();
+      assert.isTrue(adapterManager._watchMozBluetoothOnattributechanged.called);
+      assert.isTrue(adapterManager._initDefaultAdapter.called);
+    });
+  });
+
+  suite('_watchMozBluetoothOnattributechanged > ', function() {
+    var spyMockNavigatorBluetoothAddEventListener, eventName, mockEvent;
+    setup(function() {
+      spyMockNavigatorBluetoothAddEventListener =
+        this.sinon.spy(this.MockNavigatorBluetooth, 'addEventListener');
+      eventName = 'onattributechanged';
+      mockEvent = {
+        attrs: ['defaultAdapter']
+      };
+    });
+
+    teardown(function() {
+      spyMockNavigatorBluetoothAddEventListener.reset();
```

Comment:
This is not required as spies created using `this.sinon.spy` are restored automatically.